nixos/modules/services/web-apps/keycloak.nix

```diff
@@ -0,0 +1,71 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.keycloak;
+in {
+  options = {
+    services.keycloak = {
+      enable = mkEnableOption "Keycloak identity and access server";
+
+      package = mkOption {
```

Comment:
I don`t think this option is necessary. There aren't multiple variants of the `keycloak` package available in nixpkgs, and you can use an overlay to replace the `keycloak` with a different version.

Reply:
While a package option isn't necessary, I've found it a bit easier for people to grasp/use than overlays, and not uncommon in modules.
I'd prefer to keep it, for the time being..

Reply:
If you think it is reasonable that there might be multiple versions (maybe a LTS, or whatever) then that is fine :+1: